avocado/utils/network/interfaces.py

```diff
@@ -144,16 +144,37 @@ def bring_up(self):
         except Exception as ex:
             raise NWException("Failed to bring up: %s" % ex)
 
-    def is_link_up(self):
-        """Check if the interface is up or not.
+    def is_admin_link_up(self):
+        """Check the admin link state is up or not.
 
-        :return: True or False. True if the current state is UP, otherwise
-                 will return False.
+        :return: True or False, True if network interface state is 'up'
+                 or 'unknown' otherwise will return False.
         """
-        if self.get_link_state() == 'up':
+        if self.get_link_state() == 'up' or 'unknown':
```

Comment:
This PR is almost ready, thanks for the changes.

My last comment is here. `unknown` is part of the operational state. Should not be checked here. If the state is `unknown` we *cannot* say that admin link is up. Right?

Reply:
@beraldoleal but we are handling this for specific interface like veth(virtual interface), we are facing the issue veth interface thats why we included the operational state to check to whether the interface is up or down.
```
env3: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 2000 qdisc fq_codel state UNKNOWN group default qlen 1000
    link/ether 8e:b2:5d:3a:1d:03 brd ff:ff:ff:ff:ff:ff
    inet 100.10.10.204/24 scope global env3
       valid_lft forever preferred_lft forever
```

Reply:
@smruti77 Can you point the RedHat documentation here so that it is easier to review the changes ?

Reply:
My question is: Can we say that *every* interface with `UNKNOWN` state is *administratively up*? Can we ensure that? IMO, `unknown`

Now that you have the second method, asking if the operational state is up is not enough for your tests?

Reply:
@beraldoleal Previously we only checked admin state. When trying to solve the case of 'unknown', we realized that we need to check both operational and admin state.

Reply:
@narasimhan-v @smruti77 RFC2863: *operational states*. IMO, this is not part of *admin state*.

https://tools.ietf.org/html/rfc2863